internal/types/compare_order.go

```diff
@@ -91,9 +92,35 @@ const (
 	Descending SortType = -1
 )
 
+// numberOrderResult represents the comparison order of numbers.
+type numberOrderResult uint8
+
+const (
+	_ numberOrderResult = iota
+	doubleDT
+	int32DT
+	int64DT
+)
+
+// detectNumberType returns a sequence for float64, int32 and int64 types.
```

Comment:
What does it mean "a sequence", I see `numberOrderResult` is a `uint8`, is there a way we interpret it as a sequence?